### PR TITLE
Don't display deleted events in category contents

### PR DESCRIPTION
**Pull request by nop33**
Fixes a bug report.

### CODE REVIEW

File: indico/MaKaC/services/implementation/category.py [hunk 1]
```diff
@@ -159,7 +159,8 @@ def _getAnswer( self ):
         skip = max(0, self._lastIdx - 100)
         pastEvents = {}
         num = 0
-        for event in islice(index.itervalues(), skip, self._lastIdx):
+        events = [e for e in index.itervalues() if not e.as_event.is_deleted]
```
Comment (ThiefMaster) [inline indico/MaKaC/services/implementation/category.py]: That's going to send a SQL query for each event :/

Comment (nop33) [inline indico/MaKaC/services/implementation/category.py]: Isn't there any other way to check if an event is deleted? I couldn't find any related property of the legacy event object..

Comment (ThiefMaster) [inline indico/MaKaC/services/implementation/category.py]: no, a legacy event object doesn't know that it's deleted - but `ConferenceHolder().hasKey(event_id)` will return 0 if it's not in there anymore

**Comment by ThiefMaster** [comment]
It's better to fix this in the DB when it occurs: https://gist.github.com/ThiefMaster/da8cb09e874f10bb64e5

**Pull request closed without merging**